exist-core/src/main/java/org/exist/storage/NativeBroker.java

```diff
@@ -740,13 +740,15 @@ public Collection getOrCreateCollection(final Txn transaction, final XmldbURI na
             final CollectionCache collectionCache, final Optional<Tuple2<Permission, Long>> creationAttributes)
             throws TriggerException, ReadOnlyException, PermissionDeniedException, LockException, IOException {
 
-        final CollectionTrigger trigger;
-        if(parentCollection == null) {
-            trigger = new CollectionTriggers(this, transaction);
-        } else {
-            trigger = new CollectionTriggers(this, transaction, parentCollection);
+        CollectionTrigger trigger = null;
```

Comment:
is the removal of final of line 743 needed? also is the null initialisation needed?